internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/fastclient/grpc/VeniceGrpcEndtoEndTest.java

```diff
@@ -0,0 +1,155 @@
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.specific.SpecificRecord;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+
+public class VeniceGrpcEndtoEndTest extends AbstractClientEndToEndSetup {
+  private VeniceClusterWrapper cluster;
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    Utils.thisIsLocalhost();
+
+    Properties props = new Properties();
+    props.setProperty(PARTICIPANT_MESSAGE_STORE_ENABLED, "false");
+    props.setProperty(CONTROLLER_AUTO_MATERIALIZE_META_SYSTEM_STORE, "true");
+    props.put(SERVER_HTTP2_INBOUND_ENABLED, "true");
+    props.put(SERVER_QUOTA_ENFORCEMENT_ENABLED, "true");
+
+    cluster = ServiceFactory.getVeniceCluster(
+        new VeniceClusterCreateOptions.Builder().numberOfControllers(1)
+            .numberOfPartitions(1)
+            .numberOfRouters(1)
+            .numberOfServers(1)
+            .sslToStorageNodes(true)
+            .extraProperties(props)
+            .build());
+  }
+
+  @AfterClass
+  public void cleanUp() {
+    Utils.closeQuietlyWithErrorLogged(cluster);
+  }
+
+  public String writeSyntheticData(String storeName) throws IOException {
+    // 1. Create a new store in Venice
+    String uniqueStoreName = Utils.getUniqueString(storeName);
+    NewStoreResponse response = cluster.getNewStore(uniqueStoreName);
+    UpdateStoreQueryParams params = new UpdateStoreQueryParams().setStorageQuotaInByte(Store.UNLIMITED_STORAGE_QUOTA);
+
+    Assert.assertEquals(response.getName(), uniqueStoreName);
+    Assert.assertNull(response.getError());
+
+    ControllerResponse updateStoreResponse = cluster.updateStore(uniqueStoreName, params);
+    Assert.assertNull(updateStoreResponse.getError());
+
+    // 2. Write Synthetic data to the store w/ writeSimpleAvroFileWithUserSchema
+    File inputDir = TestWriteUtils.getTempDataDirectory();
+    String inputDirPath = "file://" + inputDir.getAbsolutePath();
+    TestWriteUtils.writeSimpleAvroFileWithUserSchema(inputDir);
+
+    // 3. Run a push job to push the data to Venice (VPJ)
+    Properties vpjProps = TestWriteUtils.defaultVPJProps(cluster.getRandomRouterURL(), inputDirPath, uniqueStoreName);
+    TestWriteUtils.runPushJob("test push job", vpjProps);
+
+    cluster.useControllerClient(
+        controllerClient -> TestUtils.waitForNonDeterministicAssertion(60, TimeUnit.SECONDS, () -> {
+          StoreResponse storeResponse = controllerClient.getStore(uniqueStoreName);
+          Assert.assertEquals(storeResponse.getStore().getCurrentVersion(), 1);
+        }));
+
+    return uniqueStoreName;
+  }
+
+  @Test
+  public void testThinClient() throws Exception {
```

Comment:
Let's create one test in which we verify the data first using thin client and then fast client